FoodPin/FoodPin/Controller/RestaurantTableViewController.swift

```diff
@@ -110,4 +107,82 @@ class RestaurantTableViewController: UITableViewController {
         // Deselect the row
         tableView.deselectRow(at: indexPath, animated: false)
     }
+
+    // Swipe left action
+    override func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        // Get the selected restaurant
+        guard let restaurant = self.dataSource.itemIdentifier(for: indexPath)
+        else {
+            return UISwipeActionsConfiguration()
+        }
+
+        // Delete action
+        let deleteAction = UIContextualAction(style: .destructive, title: "Delete") { (_, _, completionHandler) in
+
+            var snapshot = self.dataSource.snapshot()
+            snapshot.deleteItems([restaurant])
+            self.restaurants.remove(at: indexPath.row)
+            self.dataSource.apply(snapshot, animatingDifferences: true)
+
+            // Call completion handler to dismis the action button
+            completionHandler(true)
+        }
+
+        // Share action
+        let shareAction = UIContextualAction(style: .normal, title: "Share") {(_, _, completionHandler) in
+            let defaultText = "Just checking in at" + restaurant.name
+            let activityController: UIActivityViewController
+
+            if let imageToShare = UIImage(named: restaurant.image) {
+                activityController = UIActivityViewController(activityItems: [defaultText, imageToShare], applicationActivities: nil)
+            } else {
+                    activityController = UIActivityViewController(activityItems: [defaultText], applicationActivities: nil)
+                }
+
+            if let popoverController = activityController.popoverPresentationController {
+                if let cell = tableView.cellForRow(at: indexPath) {
+                    popoverController.sourceView = cell
+                    popoverController.sourceRect = cell.bounds
+                }
+            }
+
+            self.present(activityController, animated: true, completion: nil)
+            completionHandler(true)
+        }
+
+        deleteAction.backgroundColor = UIColor.systemRed
+        deleteAction.image = UIImage(systemName: "trash")
+
+        shareAction.backgroundColor = UIColor.systemOrange
+        shareAction.image = UIImage(systemName: "square.and.arrow.up")
+
+        // Configure both actions as swipe action
+        let swipeConfiguration = UISwipeActionsConfiguration(actions: [deleteAction, shareAction])
+        return swipeConfiguration
+    }
+
+    // Swipe right action
+    override func tableView(_ tableView: UITableView, leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        // Get the selected restaurant
+        let _: UIActivityViewController
```

Comment:
what is this used for?

Reply:
I used these to add a check-in action when we swipe right the cell. I add the swipe to favorite action.